packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,153 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param rewriteTo the value to substitute for the matched part of the path of the gateway request URL
+   *   As a default, retains original request's URL path.
+   */
+  public static regex(regex: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex }, rewriteTo);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly rewriteTo?: string,
+  ) {
+    super();
+
+    if (prefixPathMatch[0] !== '/') {
+      throw new Error('Prefix path for the match must start with \'/\', '
+        + `got: ${prefixPathMatch}`);
+    }
+
+    if (rewriteTo && rewriteTo !== '') {
+      if (prefixPathMatch[prefixPathMatch.length - 1] !== '/') {
+        throw new Error('When prefix path for the rewrite is specified, prefix path for the match must end with \'/\', '
+          + `got: ${prefixPathMatch}`);
+      }
+      if (rewriteTo[0] !== '/' || rewriteTo[rewriteTo.length - 1] !== '/') {
+        throw new Error('Prefix path for the rewrite must start and end with \'/\', '
+          + `got: ${rewriteTo}`);
+      }
+    }
+  }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefixPathMatch,
+      prefixPathRewrite: this.rewriteTo === '' || this.rewriteTo
+        ? {
+          defaultPrefix: this.rewriteTo === '' ? 'DISABLED' : undefined,
+          value: this.rewriteTo === '' ? undefined : this.rewriteTo,
+        }
+        : undefined,
+    };
+  }
+}
+
+class HttpGatewayRouteWholePathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly wholePathMatch: CfnGatewayRoute.HttpPathMatchProperty,
+    private readonly exactPathRewrite?: string | undefined,
+  ) {
+    super();
+
+    if (wholePathMatch.exact && wholePathMatch.exact[0] !== '/') {
+      throw new Error(`Exact Path for the match must start with \'/\', got: ${ wholePathMatch.exact }`);
+    }
+    if (exactPathRewrite === '') {
+      throw new Error('Exact Path for the rewrite cannot be empty. Unlike startsWith() method, no automatic rewrite on whole path match');
+    }
+    if (exactPathRewrite && exactPathRewrite[0] !== '/') {
+      throw new Error(`Exact Path for the rewrite must start with \'/\', got: ${ exactPathRewrite }`);
+    }
+  }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      wholePathMatch: this.wholePathMatch,
+      wholePathRewrite: this.exactPathRewrite
+        ? {
+          exact: this.exactPathRewrite,
+        }
+        : undefined,
```

Comment:
I understand `this. exactPathRewrite` can never be `''` here because of the checks in the constructor, but let's be explicit here anyway:

```suggestion
      wholePathRewrite: this.exactPathRewrite === undefined ? undefined : { exact: this.exactPathRewrite },
```

Reply:
well noted.